Description: fix lupdate/lrelease execution error workaround in qemu-user
  refer:
    https://mooody.me/posts/2022-04/qprocess-in-qemu/
    https://codereview.qt-project.org/c/qt/qtbase/+/313894
    https://github.com/felixonmars/archriscv-packages/pull/1151#issuecomment-1100954643
Author: Han Gao <gaohan@iscas.ac.cn>
Last-Update: 2022-09-24

--- qtbase-opensource-src-5.15.3.1.orig/src/corelib/configure.json
+++ qtbase-opensource-src-5.15.3.1/src/corelib/configure.json
@@ -1087,6 +1087,11 @@ Mozilla License) is included. The data i
             "section": "Utilities",
             "output": [ "publicFeature" ]
         },
+	"forkfd_pidfd": {
+            "label": "CLONE_PIDFD support in forkfd",
+            "condition": "config.linux",
+            "output": [ "privateFeature" ]
+        },
         "win32_system_libs": {
             "label": "Windows System Libraries",
             "condition": "config.win32 && libs.advapi32 && libs.gdi32 && libs.kernel32 && libs.netapi32 && libs.ole32 && libs.shell32 && libs.uuid && libs.user32 && libs.winmm && libs.ws2_32"
@@ -1180,7 +1185,12 @@ Please apply the patch corresponding to
                     "condition": "config.qnx"
                 },
                 "pcre2",
-                "system-pcre2"
+                "system-pcre2",
+                {
+                    "type": "feature",
+                    "args": "forkfd_pidfd",
+                    "condition": "config.linux"
+                }
             ]
         }
     ]
--- qtbase-opensource-src-5.15.3.1.orig/src/corelib/io/qprocess_unix.cpp
+++ qtbase-opensource-src-5.15.3.1/src/corelib/io/qprocess_unix.cpp
@@ -462,6 +462,12 @@ void QProcessPrivate::startProcess()
     int ffdflags = FFD_CLOEXEC;
     if (typeid(*q) != typeid(QProcess))
         ffdflags |= FFD_USE_FORK;
+
+    // QTBUG-86285
+#if !QT_CONFIG(forkfd_pidfd)
+    ffdflags |= FFD_USE_FORK;
+#endif
+
     pid_t childPid;
     forkfd = ::forkfd(ffdflags , &childPid);
     int lastForkErrno = errno;
